src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/move_readvalue_inputs_to_subgraph.cpp

```diff
@@ -0,0 +1,187 @@
+        }
+
+        // TODO: Temporarily skip this pattern. If MemoryInputSDPA supports Subgraph in the future, it may be deleted.
+        if (skip_pattern_sdpa(readvalue)) {
+            return false;
+        }
+
+        bool found_output = false;  // Flag: find Output node
+
+        NodeVector subgraph_nodes;
+        std::unordered_set<std::string> subgraph_node_names;
+        std::unordered_set<std::shared_ptr<ov::Node>> visited;  // Visited flag.
+        NodeVector inputs = {};
+        OutputVector outputs = {};
+
+        // DFS, Check if current node's final successor is only ReadValue.
+        std::function<void(std::shared_ptr<ov::Node>)> dfs = [&](std::shared_ptr<ov::Node> node) {
+            if (visited.find(node) != visited.end()) {
+                return;
+            }
+
+            // Output node.
+            if (node->get_output_target_inputs(0).size() == 0u) {
+                found_output = true;
+                return;
+            }
+
+            for (const auto& child : node->get_output_target_inputs(0)) {
+                auto son = child.get_node()->shared_from_this();
+                if (son == readvalue) {
+                    continue;
+                }
+                dfs(son);
+            }
+            if (!found_output) {
+                visited.insert(node);
+            }
+        };
+
+        std::function<void(std::shared_ptr<ov::Node>)> reverse_dfs = [&](std::shared_ptr<ov::Node> node) {
+            if (ov::op::util::is_parameter(node)) {
+                inputs.emplace_back(node);
+                return;
+            }
+
+            // Check whether current node have same successor[root_node_name].
+            found_output = false;
+            dfs(node);
+            visited.insert(node);
+            if (found_output) {
+                inputs.emplace_back(node);
+                return;
+            }
+
+            for (size_t i = 0; i < node->get_input_size(); i++) {
+                reverse_dfs(node->get_input_node_shared_ptr(i));
+            }
+
+            // Cache to subgraph_nodes
+            subgraph_nodes.emplace_back(node);
+            subgraph_node_names.insert(node->get_friendly_name());
+        };
+
+        // Reverse DFS ReadValue, find all suitable nodes and move them to subgraph_nodes.
+        reverse_dfs(readvalue->get_input_node_shared_ptr(0));
+
+        if (inputs.size() == 0 || subgraph_nodes.size() == 0) {
+            return false;
+        }
+
+        // Subgraph's input
+        auto params = ParameterVector{};
+        for (auto inp : inputs) {
+            auto param = std::make_shared<ov::op::v0::Parameter>(inp->get_element_type(), inp->get_output_partial_shape(0));
+            params.push_back(param);
+            for (const auto& child : inp->get_output_target_inputs(0)) {
+                if (subgraph_node_names.find(child.get_node()->shared_from_this()->get_friendly_name()) !=
+                    subgraph_node_names.end()) {
```

Comment:
Can the pointers be used instead of the names to speed up comparison?

Reply:
Done.